lib/generateClass.js

```diff
@@ -284,14 +307,222 @@ function writeToJson(def, cw) {
+
+            if (sliceGroup.slicing.discriminator.some( d => d.path == element.path )) { 
+              // element.path matches a discriminator; we found a value to match on
+
+              // this could be all sorts of things, fixedCode, fixedString, fixedWhatever, so figure out what type it is and look for fixed<Type>
+              let fixedValue;
+
+              for (const type of element.type) {
+                const typeName = type.code;
+                const fieldName = `fixed${typeName.charAt(0).toUpperCase()}${typeName.slice(1)}`; // ex. 'code' => 'fixedCode'
+
+                if (element[fieldName]) {
+                  fixedValue = element[fieldName];
+                }
+              }
+
+              if (!fixedValue) {
+                console.error('Could not identify fixed value for:');
+                console.error(element);
+              }
+
+              sliceGroup.elements[sliceGroup.currentSlice].unshift( { path: element.path, value: fixedValue } );
+            }
+          });
+        } 
+      }
+    }
+
+    // if the fhir profile ends and we still have "open" slices, finalize them here
+    while (sliceStack.length > 0) {
+      const sliceGroup = sliceStack.shift();
+      sliceMap[sliceGroup.path] = sliceGroup.elements;
+    }
+
+    for (let element of fhirProfile.snapshot.element) {
+      const mapping = element.mapping && element.mapping.find(m => m['identity'] === 'shr');
+      let baseIsList = element.base && element.base.max == '*';
+      baseIsList = baseIsList || (element.mapping && element.mapping.filter(elem => elem.identity == 'shr').length > 1);
+      if(mapping !== undefined){
+        if(mapping.map === '<Value>'){
+          // Mapping to the value of this es6 instance
+          if (def.value instanceof IdentifiableValue && def.value.identifier.isPrimitive) {
+            generateFromFHIRAssignment(def.value.card.isList, baseIsList, 1, element, [], 'value', fhirProfile, null, cw);
+          } else {
+            console.error('ERROR: Value referenced in mapping but none exist on this element.');
+          }
+        } else {
+          // Mapping to a field within this es6 instance
+          const fieldMapPath = mapping.map.match(/<([^>]*)>/g);
+          if(fieldMapPath){
+            // Generate a chain if mapped multiple levels deep, such as myField1.subField2
+            const classMethodChain = fieldMapPath.map(e => toSymbol(fhirMappingToIdentifier(e).name)).join('.');
+            // find related field if available
+            let field;
+            let currDef = def;
+
+            // for nested element mappings, such as "Patient.person.address", we have to look up Patient.person and then Person.address
+            for(const pathElement of fieldMapPath) {
+              field = currDef.fields.find(f => f.identifier && f.identifier.equals(fhirMappingToIdentifier(pathElement)));
+
+              if (!field) {
+                break;
+              }
```

Comment:
In researching this, I found a couple quick fixes to find the field here, specifically it could be the `value`. But there are still a lot of instances where the matching field is not found. 

My sense here is that there shouldn't be cases where the field isn't found, but I'm not yet sufficiently well-versed in the SHR/CIMPL ecosystem to resolve or explain the remaining cases. So rather than continue down that rabbit hole at the moment, I'm going to leave the `break` but add some error logging. Will follow up on this one once I've addressed the other open items.